frontend/README.md

```diff
@@ -0,0 +1,151 @@
+# Translate.Next — a better Translate app for Pontoon
```

Comment:
I like that you put code documentation in here!

For the sake of consistency, could you please also (as part of this PR, otherwise we'll forget) migrate existing code documentation from RTD?

http://mozilla-pontoon.readthedocs.io/en/latest/dev/sync.html -> /pontoon/api/README.md
http://mozilla-pontoon.readthedocs.io/en/latest/dev/api.html -> /pontoon/api/README.md